tools/refguide_check.py

```diff
@@ -835,18 +823,12 @@ def flush(self):
             # Process our options
             if any([SKIPBLOCK in ex.options for ex in t.examples]):
                 continue
-            fails, successes = runner.run(t, out=out, clear_globs=False)
+            fails, successes = runner.run(t, out=output.write, clear_globs=False)
             if fails > 0:
                 success = False
             ns = t.globs
-    finally:
-        sys.stderr = old_stderr
-        os.chdir(cwd)
-        shutil.rmtree(tmpdir)
-        np.set_printoptions(**old_printoptions)
-        np.seterr(**old_errstate)
 
-    return success, output
+    return success, output.read()
```

Comment:
output is now a str instead of array but this method starts with _ so I think this is fine.